examples/dataflow-python-examples/batch-examples/cookbook-examples/cloudbuild.yaml

```diff
@@ -0,0 +1,16 @@
+steps:
```

Comment:
please add a Step 0: docker pull / cache to speed up these builds
https://cloud.google.com/cloud-build/docs/speeding-up-builds#using_a_cached_docker_image

note that cloud build zero-indexes it's steps so as-is these comments are off-by one.

Reply:
--cached-from will be useful if we are persisting the image across builds. In the above example the **images** part pushes the image to GCR but in our case we are not persisting these images at the end of the build.  

Adjusted the step id indexes in comments to start with 0

Reply:
Is there a reason to not persist images across builds? this could speed up later runs.
For example:
PR 1 new example  -> builds test image for that example and pushes to GCR
PR 2 edit anything about the example without touching dependencies -> benefits from image being in GCR already

Reply:
We can persist images as long as everybody is in agreement and do not have any concerns on the number and size of images.